snapshots/overlay/overlay.go

```diff
@@ -102,22 +130,92 @@ func NewSnapshotter(root string, opts ...Opt) (snapshots.Snapshotter, error) {
 		return nil, err
 	}
 
-	if err := os.Mkdir(filepath.Join(root, "snapshots"), 0700); err != nil && !os.IsExist(err) {
-		return nil, err
+	firstSnapshot := true
+	if err := os.Mkdir(filepath.Join(root, "snapshots"), 0700); err != nil {
+		if !os.IsExist(err) {
+			return nil, err
+		}
+		firstSnapshot = false
 	}
 	// figure out whether "userxattr" option is recognized by the kernel && needed
 	userxattr, err := overlayutils.NeedsUserXAttr(root)
 	if err != nil {
 		logrus.WithError(err).Warnf("cannot detect whether \"userxattr\" option needs to be used, assuming to be %v", userxattr)
 	}
 
+	var (
+		metacopyOn   bool
+		metacopyCtrl bool
+	)
+	metacopyCtrl = config.metacopyCtrl && supportsMetacopy()
```

Comment:
Also, I think this causes some "bugs". If the user didn't enable metacopy in the config.toml, then this will always be false.

This means that:
1. A user enables metacopy
2. Then creates an snapshot
3. Disables metacopy
4. Can't use this snapshot

IMHO, what makes sense is that: if the file in the snapshot is present signaling it needs metacopy, we should enable it (maybe warn too). I don't think we should invalidate all snapshots taken with metacopy on if the user later wants to disable metacopy.

They probably want to disable whenever it can be disabled, but not invalidate all snapshots taken with metacopy=on